### PR TITLE
[2.x] Round prices before multiplying by qty

### PR DESCRIPTION
**Pull request by Jade-GG**
Magento rounds prices to 2 decimals before they get multiplied by qty. Not doing this can cause some slightly incorrect prices to be shown on the PDP, especially in combination with tier price discounts.

(More specifically, it converts to the given currency before multiplying by qty by calling `currencyByStore`. This isn't always 2 decimals, see [the ISO 4217 specification](https://en.wikipedia.org/wiki/ISO_4217#List_of_ISO_4217_currency_codes)---9 currencies have more than 2 decimals, and 17 currencies have no decimals at all)

### CODE REVIEW

**Review by royduin — APPROVED** [review]

**Comment by Jade-GG** [comment]
To get around the different currencies problem we could use something like

`+(new Intl.NumberFormat('en-US', {style: 'currency', currency: config.currency}).format(price).replace(/[^0-9.]/g,''))`

to let the browser handle it, but it's incredibly ugly.

**Comment by royduin** [comment]
We already have a price filter: https://github.com/rapidez/core/blob/master/resources/js/filters.js

**Comment by Jade-GG** [comment]
> We already have a price filter: https://github.com/rapidez/core/blob/master/resources/js/filters.js

This is true, however this also uses the locale, which changes the decimal separator and doesn't let us easily convert the number back into a float 🙃

**Comment by royduin** [comment]
Waiting on a review from @indykoning

**Review by indykoning — APPROVED** [review]

**Pull request merged**
